onnxruntime/core/providers/cpu/cpu_execution_provider.cc

```diff
@@ -437,6 +437,17 @@ class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain,
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 12, uint8_t, ReduceMin);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 12, GatherND);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 12, Einsum);
+// REVIEW(codemzs): ConstEigenVectorArrayMap.cast<MLFLoat16) does not seem to be supported.
+// However these types work on GPU implementation.
+//class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 12, MLFloat16_MLFloat16, Dropout);
+//class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 12, MLFloat16_float, Dropout);
+//class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 12, MLFloat16_double, Dropout);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 12, float_MLFloat16, Dropout);
```

Comment:
I think we actually don't need MLFloat16 for CPU kernels... CPU doesn't have a good support for float16 and there is no real model that needs it.
Inference team has been trying to keep the binary size small. Could you please check with Pranav and Changming to see if we should remove MLFloat16 ones.

Reply:
@snnn @pranavsharma can you guys please chime in?

Reply:
I think adding registration by itself is insufficient.
the dropout-12 kernel implementation is located under training_ops which only gets built when training is enabled.
an inference-only build of ORT wouldn't have the implementation of dropout-12

Reply:
@jywu-msft You are correct. I noticed the build failures and it’s in my plan to move it to inference ops same way as we Slice and SliceGrad. Can you please weigh in @SherlockNoMad ‘s point?

Reply:
re: MLFloat16, our rule of thumb has been to only register types that are anticipated to be used. 
if we don't think it will be used in real models, let's not register it right now.

Reply:
thanks @jywu-msft

Reply:
> > Which model requires Dropout for inferencing?
> 
> @pranavsharma Maybe @SherlockNoMad may know? I just know Dropout started as an inference operator and then came TrainableDropout and then finally came the merger called Dropout(12)

Yeah, given that this was never registered so far for inferencing wondering which model requires it?
As for the types, again it comes down to the models that require it. They're added on a case by case basis based on model usage.

Reply:
@SherlockNoMad I have removed Float16 from CPU as per your suggestion. I have moved Dropout(12) forward ops to inference project.